sponsorship_switzerland/models/contracts.py

```diff
@@ -411,3 +412,14 @@ def _reconcile_open_amount(self):
             delay = datetime.now() + relativedelta(seconds=15)
             if invoices:
                 invoices.with_delay(eta=delay).group_or_split_reconcile()
+
+    @api.multi
+    @job(default_channel='root.recurring_invoicer')
+    @related_action(action='related_action_contract')
+    def _clean_invoices(self, since_date=None, to_date=None, keep_lines=None):
+        inv_lines = self.invoice_line_ids.filtered(lambda r: r.state == 'open')
```

Comment:
Actually you need to include paid invoices that are in the future, in case the sponsor paid in advance, because those invoices will as well be cancelled.